[GPU] bfloat16 GPU kernels can fail to link when libgcc_s does not provide bf16 conversion builtins

This might be partially related with #803, #804.

## Summary
Some environments fail to link GPU kernels that use `bfloat16` scalar operations because the generated object requires bfloat16 compiler runtime conversion helper symbols such as `__truncsfbf2`.

f.g., 
```text
$ codon build -release test/transform/kernels.codon -o ~/src/test_code/codon_optimize_cou/kernels
/usr/bin/ld: /home/swchoi/src/test_code/codon_optimize_cou/kernels.o: in function `test_scalar_types.0:0.kernel_bf16.0:0[bfloat16,std.internal.types.collections.list.List.0[bfloat16]]':
kernels.codon:(.text+0xe3f8): undefined reference to `__truncsfbf2'
/usr/bin/ld: kernels.codon:(.text+0xe411): undefined reference to `__truncsfbf2'
/usr/bin/ld: kernels.codon:(.text+0xe42a): undefined reference to `__truncsfbf2'
collect2: error: ld returned 1 exit status
error: process for 'g++' exited with status 1
```

## MRE
Minimal case is `bf16` case in `test\transform\kernels.codon`, `test_scalar_types()`.
```
    @gpu.kernel
    def kernel_bf16(x: bfloat16, out):
        out[0] = (x + bfloat16(3.0)) * bfloat16(3.0)
```

### Result
On affected environments, linking fails with missing bfloat16 compiler runtime builtin symbols:
```
undefined reference to `__truncsfbf2'
```

On other environments, the same workload succeeds. 

This is because, In those environments, the missing symbol is often provided by `libgcc_s.so.1`:
```bash
$ nm -D ~/src/codon/install/lib/codon/libgcc_s.so.1 | grep -E '__truncdfbf2|__truncsfbf2|__extendbfsf2'
0000000000014220 T __extendbfsf2@@GCC_13.0.0
0000000000018580 T __truncdfbf2@@GCC_13.0.0
0000000000018bf0 T __truncsfbf2@@GCC_13.0.0
```

`Older libgcc_s versions` (especially GCC with older than 13.0.0 versions) may not provide these symbols. For example, Ubuntu 22.04 jammy libgcc-s1 does not export these bf16 helpers:

```
$ nm -D jammy/lib/x86_64-linux-gnu/libgcc_s.so.1 \
  | grep -E '__truncdfbf2|__truncsfbf2|__extendbfsf2'

$ nm -D noble/usr/lib/x86_64-linux-gnu/libgcc_s.so.1   | grep -E '__truncdfbf2|__truncsfbf2|__extendbfsf2'
0000000000014010 T __extendbfsf2@@GCC_13.0.0
0000000000018380 T __truncdfbf2@@GCC_13.0.0
00000000000189f0 T __truncsfbf2@@GCC_13.0.0
```

So linking may fails environment by environment.

## Root Cause
But, current compiler-rt provides the required `bf16` conversion helpers, and codon links LLVM compiler rt builtins into `libcodonrt.so`:
```
nm ~/tool/llvm-project/install/lib/clang/20/lib/x86_64-unknown-linux-gnu/libclang_rt.builtins.a   | grep -E '__truncdfbf2
|__truncsfbf2|__extendbfsf2'
0000000000000000 T __extendbfsf2
0000000000000000 T __truncdfbf2
0000000000000000 T __truncsfbf2
```

Therefore, whether the link succeeds can depend on the system or bundled libgcc_s.so.1. 

Environments with newer libgcc may succeed, while environments with older libgcc may fail.

As Codon already depends on LLVM/compiler-rt, the more robust fix is to explicitly mark the bfloat16 compiler runtime builtins as default-visible when preparing libclang_rt.builtins.default.a, allowing libcodonrt.so to provide these symbols consistently.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -179,13 +179,19 @@ else()
   set(COMPILER_RT_DEFAULT "${CMAKE_BINARY_DIR}/libclang_rt.builtins.default.a")
   set(COMPILER_RT_DEFAULT_VIS_SYMBOLS
     __truncdfhf2
+    __truncdfbf2
     __truncdfsf2
+    __truncsfbf2
     __truncsfhf2
     __trunctfdf2
+    __trunctfbf2
     __trunctfhf2
     __trunctfsf2
+    __truncxfbf2
     __trunctfxf2
+    __trunchfbf2
     __truncxfhf2
+    __extendbfsf2
     __extendsfdf2
     __extendhfsf2
     __extenddftf2

